audbcards/core/dataset.py

```diff
@@ -274,6 +292,283 @@ def version_link(self) -> str:
+                [db.schemes[s].maximum is not None for s in db.schemes]
+            )
+            has_labels = any(
+                [db.schemes[s].labels is not None for s in db.schemes]
+            )
+            has_mappings = any(
+                [
+                    isinstance(db.schemes[s].labels, (str, dict))
+                    for s in db.schemes
+                ]
+            )
+
+        columns = ['ID', 'Dtype']
+        header_line = '   :header: ID,Dtype'
+        if has_minimums:
+            header_line += ',Min'
+            columns.append('Min')
+        if has_maximums:
+            header_line += ',Max'
+            columns.append('Max')
+        if has_labels:
+            header_line += ',Labels'
+            columns.append('Labels')
+        if has_mappings:
+            header_line += ',Mappings'
+            columns.append('Mappings')
+
+        scheme_info['header_line'] = header_line
+        scheme_info['has_minimums'] = has_minimums
+        scheme_info['has_maximums'] = has_maximums
+        scheme_info['has_labels'] = has_labels
+        scheme_info['has_mappings'] = has_mappings
+
+        scheme_info['columns'] = columns
+        return scheme_info
+
+    @property
+    def dataset_schemes(self) -> list:
+
+        db = self.header
+        dataset_schemes = []
+        for scheme_id in db.schemes:
+            dataset_scheme = self._scheme_to_list(scheme_id)
+            dataset_schemes.append(dataset_scheme)
+
+        cols = self.scheme_info['columns']
+        data = pd.DataFrame.from_dict(dataset_schemes)[cols]
+        filter = data.applymap(lambda d: d == [])
+        data.mask(filter, other='', inplace=True)
+        scheme_data = np.array(data).tolist()
+        return scheme_data
+
+
+class Datacard(object):
+
+    def __init__(self, dataset: Dataset):
+
+        self._dataset = dataset
+
+    @property
+    def props(self) -> dict:
+        props = self._dataset.properties()
+        return props
+
+    def _render_template(self):
+
+        t_dir = os.path.join(os.path.dirname(__file__), 'templates')
+        environment = jinja2.Environment(loader=jinja2.FileSystemLoader(t_dir),
+                                         trim_blocks=True)
+        environment.filters.update(zip=zip,
+                                   tw=self._trim_trailing_whitespace,
+                                   )
+        template = environment.get_template("datacard.j2")
+        content = template.render(self.props)
+        return content
+
+    @staticmethod
+    def _trim_trailing_whitespace(x: list):
+        """J2 filter to get rid or trailing empty table entries within a row.
+
+        Trims last entry if present.
+
+        Args:
+            x: untrimmed single scheme table row
+        Returns:
+            trimmed single scheme table row
+        """
+
+        if x[-1] == '':
+            x.pop()
+
+        return x
+
+    def save(self, ofpath=None):
+        """save content to rst."""
```

Comment:
```suggestion
        """Save content to rst."""
```

What does content refer to? The datacard?

Reply:
This is about the rendered content.  Commit f5283de describes this better.